scripts/globals/additional_effects.lua

```diff
@@ -0,0 +1,263 @@
+        local MP = 10 -- need actual calculation here!
+        attacker:addMP(MP)
+        -- We have to fake this or it will say the defender was healed rather than the attacker.
+        attacker:messageBasic(tpz.msg.basic.ADD_EFFECT_MP_HEAL)
+        -- We're faking it, so return zeros!
+        msgID = 0
+        msgValue = 0
+
+    elseif addType == procType.HP_DRAIN or (addType == procType.HPMPTP_DRAIN and math.random(1,3) == 1) then -- procType.HP_DRAIN
+        damage = damage * applyResistanceAddEffect(attacker, defender, element, 0)
+        if damage > defender:getHP() then
+            damage = defender:getHP()
+        end
+
+        msgID = tpz.msg.basic.ADD_EFFECT_HP_DRAIN
+        msgValue = damage
+        defender:addHP(-damage)
+        attacker:addHP(damage)
+
+    elseif addType == procType.MP_DRAIN or (addType == procType.HPMPTP_DRAIN and math.random(1,3) == 2) then
+        -- damage = damage * (math.random(90, 110)/100) -- Artificially forcing 20% variance.
+        damage = damage * applyResistanceAddEffect(attacker, defender, element, 0)
+        if damage > defender:getMP() then
+            damage = defender:getMP()
+        end
+        msgID = tpz.msg.basic.ADD_EFFECT_MP_DRAIN
+        msgValue = damage
+        defender:addMP(-damage)
+        attacker:addMP(damage)
+
+    elseif addType == procType.TP_DRAIN or (addType == procType.HPMPTP_DRAIN and math.random(1,3) == 3) then
+        -- damage = damage * (math.random(90, 110)/100) -- Artificially forcing 20% variance.
+        damage = damage * applyResistanceAddEffect(attacker, defender, element, 0)
+        if damage > defender:getTP() then
+            damage = defender:getTP()
+        end
+        msgID = tpz.msg.basic.ADD_EFFECT_TP_DRAIN
+        msgValue = damage
+        defender:addTP(-damage)
+        attacker:addTP(damage)
+
+    elseif addType == procType.DISPEL then
+        local dispel = defender:dispelStatusEffect()
+        -- Resistance check should be in dispelStatusEffect() itself
+        if dispel == tpz.effect.NONE then
+            return 0, 0, 0
+        else
+            msgID = tpz.msg.basic.ADD_EFFECT_DISPEL
+            msgValue = dispel
+        end
+
+    elseif addType == procType.SELF_BUFF then
+        if addStatus == tpz.effect.TELEPORT then -- WARP
+            attacker:addStatusEffectEx(tpz.effect.TELEPORT, 0, tpz.teleport.id.WARP, 0, 1)
+            msgID = tpz.msg.basic.ADD_EFFECT_WARP
+            msgValue = 0
+        elseif addStatus == tpz.effect.BLINK then -- BLINK http://www.ffxiah.com/item/18830/gusterion
+            -- Does not stack with or replace other shadows
+            if attacker:hasStatusEffect(tpz.effect.BLINK) or attacker:hasStatusEffect(tpz.effect.UTSUSEMI) then
+                return 0, 0, 0
+            else
+                attacker:addStatusEffect(tpz.effect.BLINK, power, 0, duration)
+                -- We're faking it, so return zeros!
+                msgID = 0
+                msgValue = 0
```

Comment:
`addStatusEffectEx` in combination with msgID 166 should do the trick.

Contrary to [the comment for ADD_EFFECT_STATUS_3](https://github.com/project-topaz/topaz/blob/c52431e2c48516954aa2f5ebbb331238f43cc3d8/scripts/globals/msg.lua#L240), Windower describes it as:

[166 - Additional effect: \<actor\> gains the effect of \<status\>](https://github.com/Windower/Lua/wiki/Message-IDs)

Reply:
I'll doublecheck this weekend, but i was certain that I was getting told _the mob_ was getting the effect instead of me when I tried all of them.

Reply:
you were right and I've adjusted it